include/rmm/mr/device/detail/arena.hpp

```diff
@@ -0,0 +1,616 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/detail/aligned.hpp>
+#include <rmm/detail/error.hpp>
+
+#include <cuda_runtime_api.h>
+
+#include <algorithm>
+#include <limits>
+#include <mutex>
+#include <set>
+#include <unordered_map>
+
+namespace rmm {
+namespace mr {
+namespace detail {
+namespace arena {
+
+constexpr std::size_t superblock_size = 1u << 26u;  ///< Size of a superblock (64 MiB).
```

Comment:
This looks like it should be a non-type template parameter of `arena` instead of a global.

Reply:
We do have a function `is_superblock()` in `block` itself.